src/rt/sched/backpressure.h

```diff
@@ -0,0 +1,259 @@
+ * thread running the cown or holding the muted cown in its mute map. If a
+ * priority message requires a recipient cown to be unmuted, then the scheduler
+ * thread running the message will search for the cown in its mute map. If the
+ * cown does not exist, then it will be sent to other scheduler threads until it
+ * is found and subsequently unmuted.
+ *
+ * ## Cown Backpressure States
+ *
+ * These states are not explicitly encoded, but they are useful to consider.
+ *
+ *     +------------+     +------------+
+ *  +--|            |---->|            |--+
+ *  |  | Normal     |     | Muting     |  |
+ *  +->|            |<-+  |            |<-+
+ *     +------------+  ^  +------------+
+ *          | ^        |      |
+ *          v |        |      v
+ *     +------------+  |  +------------+
+ *     |            |  |  |            |
+ *     | Overloaded |  +--| Muted      |
+ *     |            |     |            |
+ *     +------------+     +------------+
+ *
+ */
+
+#include "ds/mpscq.h"
+
+namespace verona::rt
+{
+  class Backpressure
+  {
+    static constexpr uint32_t current_load_mask = 0x00'0000'ff;
+    static constexpr uint32_t load_hist_mask = 0x00'ffff'00;
+    static constexpr uint32_t muted_mask = 0x80'0000'00;
+    static constexpr uint32_t needs_token_mask = 0x40'0000'00;
+
+    static constexpr uint32_t overload_threshold = 800;
+    static constexpr uint32_t unmute_threshold = 100;
+
+    /**
+     * Backpressure bits contain the following fields:
+     *   [31:31] muted
+     *   [30:30] needs_token
+     *   [29:24] (unused)
+     *   [23:08] load_history
+     *   [07:00] current_load
+     *
+     * The `muted` bit indicates if a cown is currently muted. A muted cown must
+     * not run nor be collected until it has been unmuted.
+     *
+     * The `needs_token` bit indicates if the token message is not in a cown's
+     * queue and that a new token message should be added if it runs another
+     * message.
+     *
+     * The 16-bit `load_history` field is a ring buffer with a capacity for 4
+     * 4-bit entries.
+     *
+     * The 8-bit load field tracks an approximation of a cown's message queue
+     * depth as a count of messages processed between token messages falling out
+     * of the queue. When the token message falls out, the `reset_load()`
+     * function will push the upper nibble of the current load into the load
+     * history ring buffer.
+     */
+    uint32_t bits = 0 | needs_token_mask;
+
+  public:
+    /**
+     * Return true if this cown is muted.
+     */
+    inline bool muted() const
+    {
+      return (bits & muted_mask) != 0;
+    }
+
+    /**
+     * Return true if the token message is not in this cown's queue and a new
+     * token message should be added if it runs another message.
+     */
+    inline bool needs_token() const
+    {
+      return (bits & needs_token_mask) != 0;
+    }
+
+    /**
+     * Return the amount of messages processed since the last token message fell
```

Comment:
is amount here the number? 

"Return the number of messages"?

Is this message very current implementation specific? This could change later right? It's getting a representation of the load that happens to be X right now?

If that is the case, I think being less specific in this comment and more along the line s of "get current load" would be better.